app/services/journey_logger_service.rb

```diff
@@ -5,8 +5,13 @@ def call(assessment_id, calculation_result, check, portal_user_office_code, cook
 
       attributes = build_attributes(calculation_result, check, portal_user_office_code)
       CompletedUserJourney.transaction do
-        if (journey = CompletedUserJourney.find_by(assessment_id:))
-          journey.update!(attributes)
+        existing_journey_with_early_check = CompletedUserJourney.find_by(assessment_id:, early_eligibility_result: true)
```

Comment:
this scoping looks good, but there are a couple more things to make this work:
- we need to make sure we only look to update an existing early check if the check itself is an early check, you can use the check object to work this out 
- we need to scope the existing full checks with `early_ineligible_result: false` as well as the assessment id. find_by will just find the first instance of something, so it might find the wrong result if we don't add the additional scope.

Oherwise this looks like it is on the right track!